amr-wind/turbulence/LES/Smagorinsky.cpp

```diff
@@ -52,9 +63,20 @@ void Smagorinsky<Transport>::update_turbulent_viscosity(
             const auto& mu_arr = mu_turb(lev).array(mfi);
             const auto& rho_arr = den(lev).const_array(mfi);
 
+            const auto& vof_arr = vof(lev).const_array(mfi);
+
             amrex::ParallelFor(
                 bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                    const amrex::Real rho = rho_arr(i, j, k);
+                    amrex::Real rho;
+
+                    const bool is_air = vof_arr(i, j, k) < 1.;
+                    //
+                    if (is_vof) {
```

Comment:
I think I know now why this didn't work! you're using the minimum density whenever the vof variable is present (is_vof). You meant to do that only when there is air in the current cell (is_air).

Reply:
This has been updated in the latest version.